File: wellnest-backend/src/main/java/com/wellnest/service/HealthTipService.java
package com.wellnest.service;

import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Map;
import java.util.concurrent.ThreadLocalRandom;

@Service
public class HealthTipService {

    private static final List<Map<String, String>> HEALTH_TIPS = List.of(
            Map.of(
                    "tip", "Drink enough water through the day; clear or light-yellow urine usually means you are well hydrated.",
                    "source", "WellNest Health Library"
            ),
            Map.of(
                    "tip", "Aim for 7-9 hours of sleep nightly to support recovery, immunity, and mental focus.",
                    "source", "WellNest Health Library"
            ),
            Map.of(
                    "tip", "Fill at least half your plate with vegetables and fruit to improve fiber, vitamins, and mineral intake.",
                    "source", "WellNest Health Library"
            ),
            Map.of(
                    "tip", "Include protein in each main meal to help preserve muscle mass and improve satiety.",
                    "source", "WellNest Health Library"
            ),
            Map.of(
                    "tip", "Take a 5-minute movement break each hour if you sit for long periods to reduce stiffness and improve circulation.",
                    "source", "WellNest Health Library"
            ),
            Map.of(
                    "tip", "Limit highly processed foods and added sugars; prioritize whole foods for better long-term metabolic health.",
                    "source", "WellNest Health Library"
            ),
            Map.of(
                    "tip", "Do at least 150 minutes of moderate aerobic activity weekly and include strength training at least twice per week.",
                    "source", "WellNest Health Library"
            ),
            Map.of(
                    "tip", "Track your blood pressure periodically if possible; early detection helps prevent heart-related complications.",
                    "source", "WellNest Health Library"
            ),
            Map.of(
                    "tip", "Manage stress with breathing, walking, or mindfulness for a few minutes daily to support heart and mental health.",
                    "source", "WellNest Health Library"
            ),
            Map.of(
                    "tip", "Protect your posture: keep screens at eye level and shoulders relaxed to reduce neck and back strain.",
                    "source", "WellNest Health Library"
            )
    );

    public Map<String, String> getRandomTip() {
        int index = ThreadLocalRandom.current().nextInt(HEALTH_TIPS.size());
        return HEALTH_TIPS.get(index);
    }
}

File: wellnest-frontend/src/services/healthTipService.js
import api from './api';

const fallbackTips = [
  'Drink enough water during the day to support energy, digestion, and temperature regulation.',
  'Aim for 7-9 hours of sleep each night to improve recovery and mental focus.',
  'Add vegetables and fruits to most meals for better fiber, vitamins, and mineral intake.',
  'Do at least 30 minutes of moderate movement daily, such as brisk walking or cycling.',
  'Take short movement and stretch breaks every hour if you sit for long periods.',
];

const healthTipService = {
  getRandomTip: async () => {
    try {
      const response = await api.get('/health-tips/random');
      const tip = response?.data?.data?.tip;
      const source = response?.data?.data?.source || 'WellNest Health Library';

      if (tip && typeof tip === 'string') {
        return {
          success: true,
          data: {
            tip,
            source,
          },
        };
      }
    } catch (error) {
      console.error('Failed to fetch random health tip:', error);
    }

    const randomFallback = fallbackTips[Math.floor(Math.random() * fallbackTips.length)];
    return {
      success: true,
      data: {
        tip: randomFallback,
        source: 'WellNest Health Library',
      },
    };
  },
};

export default healthTipService;
